pkg/chunked/cache_linux.go

```diff
@@ -154,6 +161,23 @@ fallback:
 	return buf, nil, err
 }
 
+func makeBinaryDigest(stringDigest string) ([]byte, error) {
+	d, err := digest.Parse(stringDigest)
+	if err != nil {
+		return nil, err
+	}
+	digestBytes, err := hex.DecodeString(d.Encoded())
+	if err != nil {
+		return nil, err
+	}
+	algo := []byte(d.Algorithm())
+	buf := make([]byte, len(algo)+1+len(digestBytes))
```

Comment:
```suggestion
	buf := make([]byte, 0, len(algo)+1+len(digestBytes))
```
otherwise this produces double-length output, with first half all zeroes.

Reply:
(Maybe worth writing a tiny unit test … it’s not much more work than testing manually and throwing the test away.)

Reply:
ouch.  This is embarrassing, thanks for catching it!

Pushed a new version, and a test